index.js

```diff
@@ -63,6 +61,38 @@ module.exports.extra = function (pkg, deps, options) {
   return missing
 }
 
+function getDetective (name) {
+  try {
+    return name
+      ? (typeof name === 'string' ? require(name) : name)
+      : require('detective')
+  } catch (e) {}
+}
+
+function noopDetective () {
+  return []
+}
+
+function getExtensions (extensions, detective) {
+  var result = {
+    '.js': getDetective(detective),
+    '.node': noopDetective,
+    '.json': noopDetective
+  }
```

Comment:
Yes, definitely, it was just easier to write the first pass like this. I will fix this up. 

> The addition of .node and .json here is so that the built-in detective shouldn't warn about them?

`.json` is an existing feature. `.node` is a new addition, but I can remove it to stay consistent. In https://github.com/maxogden/dependency-check/issues/26#issuecomment-113969574 it was mentioned that it should support whatever node.js supports by default, so that would include `.node`. You can see later in the code changes where I deleted the `.json` extension hack in favour of using the official resolution.